sdks/agent-sdk/src/core/Agent.ts

```diff
@@ -110,12 +142,22 @@ export class Agent<ContentTypes> extends EventEmitter<
     return new Agent({ client });
   }
 
-  use(middleware: AgentMiddleware<ContentTypes>) {
-    this.#middleware.push(middleware);
+  use(
+    ...middleware: Array<
+      AgentMiddleware<ContentTypes> | AgentMiddleware<ContentTypes>[]
+    >
+  ): this {
```

Comment:
so this allows passing an array of arrays?

Reply:
It allows us to do `agent.use([mw1, mw2], [mw3], mw4);`, but it does not allow `agent.use([[mw1, mw2]], [mw3], mw4)`.